### DISCUSSION TITLE
Python/Mathopt: access to Gurobi's addGenConstrIndicator

### DISCUSSION THREAD

**Discussion by quentinplsrd** [google/or-tools#4403] [Linear Solver questions] [answered]
In the current version of python mathopt, is there a way to generate an [indicator constraint](https://www.gurobi.com/documentation/current/refman/py_model_agc_indicator.html)?
Is it something that might be implemented in the future?

**Comment by sschnug** [comment, marked as answer]
I guess so. See:

- [Model](https://github.com/google/or-tools/blob/391c782dd9b93d3d9ce2f02113c25671d87e8c07/ortools/math_opt/model.proto#L199)
  - `message IndicatorConstraintProto`
  - or [C++ API](https://github.com/google/or-tools/blob/391c782dd9b93d3d9ce2f02113c25671d87e8c07/ortools/math_opt/cpp/model.h#L613)
- [Gurobi-Wrapper](https://github.com/google/or-tools/blob/391c782dd9b93d3d9ce2f02113c25671d87e8c07/ortools/math_opt/solvers/gurobi/g_gurobi.h#L402)
  - `AddIndicator(...)`

(There are surely some steps in-between one should check to make sure the information of the model is passed on to gurobi)

**EDIT:** Oops. Did not see your *python* requirement. It [*"looks"*](https://github.com/google/or-tools/blob/ed94162b910fa58896db99191378d3b71a5313af/ortools/math_opt/python/model_test.py) to me that this is [not wrapped yet](https://github.com/search?q=repo%3Agoogle%2For-tools%20path%3A%2F%5Eortools%5C%2Fmath_opt%5C%2Fpython%5C%2F%2F%20indicator&type=code) but it will surely land eventually.

**Reply by rma350** [reply]
Correct, this is not yet wrapped, we are actively working on restructuring the Python code to support the full API.
